row_cache.hh

```diff
@@ -179,6 +179,8 @@ public:
         // Updaters that are exception safe should handle all exceptions in prepare
         // Any exceptions from execute() are fatal.
         virtual void execute() = 0;
+        // Cleanup must never fail.
+        virtual future<> cleanup() noexcept { return make_ready_future<>(); }
```

Comment:
Please document what that cleanup must be executed even if prepare fails.

IMO, it's cleaner if the execution is up to the first failure: if prepare fails, it cleans up after itself. But it's okay this way too.

Reply:
cleanup was dequeued from v3